commit 0e91d09081c00664b9b7ec7040e3fdc68a8c2cc8 (refs/changes/97/4697/3)
Author: Mike Gerdts <mike.gerdts@joyent.com>
Date:   2018-08-23T17:06:14+00:00 (1 year, 2 months ago)
    
    OS-7118 imgadm support for requirements.bootrom
    Reviewed by: Pedro Palazon Candel <pedro@joyent.com>
    Reviewed by: Marsell Kukuljevic <marsell@joyent.com>
    Approved by: Marsell Kukuljevic <marsell@joyent.com>

diff --git a/src/img/node_modules/imgmanifest/lib/imgmanifest.js b/src/img/node_modules/imgmanifest/lib/imgmanifest.js
index 9272f856..bfdc99e4 100644
--- a/src/img/node_modules/imgmanifest/lib/imgmanifest.js
+++ b/src/img/node_modules/imgmanifest/lib/imgmanifest.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2016, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -1384,6 +1384,38 @@ var validators = {
         }
         delete reqs.max_platform;
 
+        if (reqs.bootrom) {
+            var VALID_BOOTROMS = ['bios', 'uefi'];
+            var brand = manifest.requirements.brand;
+            if (!brand) {
+                errs.push({
+                    field: 'requirements.bootrom',
+                    code: 'MissingParameter',
+                    message: '"requirements.brand" not specified. '
+                        + '"requirements.bootrom" requires '
+                        + '"requirements.brand" to be set to "bhyve"'
+                });
+            } else if (brand !== 'bhyve') {
+                errs.push({
+                    field: 'requirements.bootrom',
+                    code: 'Invalid',
+                    message: format('invalid bootrom '
+                        + '"requirements.bootrom" not supported with "%s" '
+                        + 'brand; "requirements.brand" must be "bhyve"', brand)
+                });
+            } else if (VALID_BOOTROMS.indexOf(reqs.bootrom) === -1) {
+                errs.push({
+                    field: 'requirements.bootrom',
+                    code: 'Invalid',
+                    message: format('invalid bootrom '
+                        + '"requirements.bootrom" invalid entry "%s"; '
+                        + 'must be one of: %s',
+                        reqs.bootrom, VALID_BOOTROMS.join(', '))
+                });
+            }
+        }
+        delete reqs.bootrom;
+
         // unknown requirements
         Object.keys(reqs).forEach(function (field) {
             errs.push({
diff --git a/src/img/node_modules/imgmanifest/package.json b/src/img/node_modules/imgmanifest/package.json
index d252ce50..c0e2384c 100644
--- a/src/img/node_modules/imgmanifest/package.json
+++ b/src/img/node_modules/imgmanifest/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgmanifest",
   "description": "Library for working with Triton/SDC/SmartOS image manifests",
-  "version": "3.0.0",
+  "version": "3.1.0",
   "author": {
     "name": "Joyent",
     "url": "joyent.com"
diff --git a/src/img/package.json b/src/img/package.json
index 8112fc9a..73a3cd43 100644
--- a/src/img/package.json
+++ b/src/img/package.json
@@ -14,7 +14,7 @@
     "docker-registry-client": "3.2.10",
     "extsprintf": "1.2.0",
     "findit2": "2.2.3",
-    "imgmanifest": "3.0.0",
+    "imgmanifest": "3.1.0",
     "mkdirp": "0.5.0",
     "node-uuid": "1.4.1",
     "once": "1.3.1",
